examples/platforms/cc2538/radio.c

```diff
@@ -390,22 +432,14 @@ void cc2538RadioProcess(otInstance *aInstance)
     if ((sState == kStateReceive && sReceiveFrame.mLength > 0) ||
         (sState == kStateTransmit && sReceiveFrame.mLength > IEEE802154_ACK_LENGTH))
     {
-#if OPENTHREAD_ENABLE_DIAG
-
-        if (otPlatDiagModeGet())
-        {
-            otPlatDiagRadioReceiveDone(aInstance, &sReceiveFrame, sReceiveError);
-        }
-        else
-#endif
+        // signal MAC layer for each received frame if promiscous is enabled
+        // otherwise only signal MAC layer for non-ACK frame
+        if (((HWREG(RFCORE_XREG_FRMFILT0) & RFCORE_XREG_FRMFILT0_FRAME_FILTER_EN) == 0) ||
+            (sReceiveFrame.mLength > IEEE802154_ACK_LENGTH))
         {
-            // signal MAC layer for each received frame if promiscous is enabled
-            // otherwise only signal MAC layer for non-ACK frame
-            if (((HWREG(RFCORE_XREG_FRMFILT0) & RFCORE_XREG_FRMFILT0_FRAME_FILTER_EN) == 0) ||
-                (sReceiveFrame.mLength > IEEE802154_ACK_LENGTH))
+            if (sReceiveDoneCallback)
             {
-
-                otPlatRadioReceiveDone(aInstance, &sReceiveFrame, sReceiveError);
+                sReceiveDoneCallback(aInstance, &sReceiveFrame, sReceiveError);
```

Comment:
@darconeous please correct me if I'm wrong but guessing the intentions of this PR, it looks like we want to, at run-time, change an NCP behavior from running OT on the chip to running OT at the host.

If my above understanding is correct, this sounds as an interesting exploration/development change, something to play around, but it raises concerns once it is used in production.